src/librustc/middle/mem_categorization.rs

```diff
@@ -727,7 +727,13 @@ impl<'a, 'gcx, 'tcx> MemCategorizationContext<'a, 'gcx, 'tcx> {
 
         let kind = match self.tables.closure_kinds().get(fn_hir_id) {
             Some(&(kind, _)) => kind,
-            None => span_bug!(span, "missing closure kind")
+            None => {
```

Comment:
nit: I think proper style would be to match on the type before trying to access `closure_kinds` with an "invalid" kind.